Assets/_Laz/Scripts/UserInterface/DebugBehaviour/DebugLazMovement/DebugMovementParametersEditor.cs

```diff
@@ -0,0 +1,312 @@
+        {
+            using (new GUILayout.VerticalScope(GUILayout.Height(_footerHeight)))
+            {
+                GUILayout.FlexibleSpace();
+                
+                DisplayFileName(_footerHeight);
+                
+                GUILayout.FlexibleSpace();
+                
+                using (new GUILayout.HorizontalScope(GUI.skin.box, GUILayout.Height(0.6f * _footerHeight)))
+                {
+                    GUILayout.FlexibleSpace();
+
+                    if (GUILayout.Button("Reset Movement", GUILayout.ExpandHeight(true), GUILayout.Width(0.45f*_rect.width)))
+                    {
+                        ResetContent();
+                    }
+
+                    GUILayout.FlexibleSpace();
+
+                    if (GUILayout.Button("Save Movement", GUILayout.ExpandHeight(true), GUILayout.Width(0.45f*_rect.width)))
+                    {
+                        SaveMovementProperty();
+                    }
+                    
+                    GUILayout.FlexibleSpace();
+                }
+                
+                GUILayout.FlexibleSpace();
+            }
+            
+        }
+        
+        //Can only create GUIStyles on GUICalls
+        //So Create GUIStyles on first call
+        private void CreateGUIStyles()
+        {
+            if (_headerGUIStyle != GUIStyle.none && _labelGUIStyle != GUIStyle.none &&
+                _valueGUIStyle != GUIStyle.none && _fileNameStyle != GUIStyle.none)
+            {
+                return;
+            }
+            
+            _headerGUIStyle = new GUIStyle(GUI.skin.label)
+            {
+                fontStyle = FontStyle.Bold,
+                alignment = TextAnchor.MiddleCenter,
+                normal = new GUIStyleState
+                {
+                    textColor = Color.black
+                },
+                stretchWidth = true,
+                stretchHeight = true,
+            };
+
+            _labelGUIStyle = new GUIStyle(GUI.skin.label)
+            {
+                fontStyle = FontStyle.Bold,
+                alignment = TextAnchor.MiddleCenter,
+                normal =  new GUIStyleState
+                {
+                    textColor = Color.white
+                }
+            };
+
+            _valueGUIStyle = new GUIStyle()
+            {
+                stretchWidth = false,
+                stretchHeight = true,
+                normal = new GUIStyleState
+                {
+                    textColor = Color.white
+                }
+            };
+
+            RectOffset fileNamePadding = GUI.skin.textField.padding;
+            fileNamePadding.top = 4;
+            
+            _fileNameStyle = new GUIStyle(GUI.skin.textField)
+            {
+                padding = fileNamePadding
+            };
+        }
+
+        private void CreateGUIParameters()
+        {
+            if (_rectTransform == null)
+            {
+                _rectTransform = GetComponent<RectTransform>();
```

Comment:
Isn't there a place where you can grab the RectTransform once and save it?  @unoctium1 @Gursaanj

Reply:
@m4rtint Isn't that what's happening here? Unless I'm reading it wrong this is just getting called once in Awake. Letting it be overridden in the inspector seems like a good call here too

Reply:
It's getting called `OnRectTransformDimensionsChange` 

So i'm assuming when the screen changes shape, or like unity editor game view changes shape.

Reply:
Ah I missed that. After playing around with it a little it does seem to only be called at runtime, so yeah I think these lines should be pulled out of CreateGUIParameters and put in Awake